velox/exec/AggregateWindow.cpp

```diff
@@ -424,7 +424,8 @@ void registerAggregateWindowFunction(const std::string& name) {
               pool,
               stringAllocator,
               config);
-        });
+        },
```

Comment:
Is this correct ?

AggregateWindow can be streaming only if the frame allows for incremental computation like say RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW. If the frame is something adhoc like ROWS BETWEEN c1 PRECEDING and c2 FOLLOWING, then the computation could involve any set of frame rows (so is PARTITION BOUND).

Reply:
@aditi-pandit Yes. I have checked the agg window function with default frame can support streaming [here](https://github.com/facebookincubator/velox/pull/9025/files#diff-20dbd8f83c46e8d4bfcb760d98ddfcc75434651eb8fc0f109056ac468ff4ace6R212).

Reply:
@JkSelf : We can set supportsSlidingFrame = true for aggregate windows as it supports incremental aggregation. Any particular reason you are not doing so ?

Reply:
@aditi-pandit We only support rank, dense_rank, row_number with all frame types, as well as agg window functions with the default frame. Here, we need a parameter to indicate whether this is an Agg window function, and then determine if it is using the default frame. If this is set to true, how do we judge in the [supportRowsStreaming ](https://github.com/facebookincubator/velox/pull/9025/files#diff-20dbd8f83c46e8d4bfcb760d98ddfcc75434651eb8fc0f109056ac468ff4ace6R213)method whether it is an agg window function with the default frame? Alternatively, we could add another parameter, isAggWindowFunction, to mark whether this function is an agg window function or not. What do you think? Thanks.

Reply:
@JkSelf : So an option is to add another enum in Scope for kFrame. So functions can be row, frame or partition scope. first_value, last_value, nth_value and aggregate functions are frame scope. But only aggregates support sliding frame.

So then in supportsRowsStreaming the conditions would be 

```
if (rows mode) {
  return true;  
}

if (frame mode and default frame and slidingframe) {
  return true;
}

return false;
```

wdyt ?

Reply:
@aditi-pandit Regarding the initial question, we introduced WindowFunctionMetadata primarily to identify which functions support row streaming. If a function supports row streaming, then its Scope should be set to kRow. For functions like Rank, dense_rank, and row_number(), we can directly set the Scope to kRow. However, for aggregate window functions, we only support the default frame, so we need to add a boolean variable to distinguish this. Personally, I think renaming isSupportSlidingFrame to isOnlySupportDefaultFrame might be more appropriate. What do you think?

Reply:
@JkSelf : Actually we can simplify this more. We can just call the boolean isAggregateWindow. Then all optimizations for aggregate windows are applicable for it. e.g. if we decide to implement a sliding algorithm, then the options can change based on it.

wdyt ?

Reply:
@aditi-pandit It makes sense to me. I have changed the name to `isAggregateWindow`.